app/controllers/api/v1/users_controller.rb

```diff
@@ -0,0 +1,30 @@
+class Api::V1::UsersController < ApplicationController
+  skip_before_action :authorized
+
+  def create
+    @user = User.create(user_params)
+    if @user.valid?
+      token = encode_token({ user_id: @user.id })
+      render json: { user: @user.username, token: token }, status: :ok
+    else
+      render json: { error: 'Invalid username' }, status: 400
+    end
+  end
+
+  def login
```

Comment:
As mentioned in my comment on routes, this isn't following proper RESTful conventions so it's better you place this login logic in its own controller with the "create" action name.